### ISSUE TITLE
`D_iterator` is tranposed compared to what documentation implies

### ISSUE THREAD

**Issue by ariaradick** [moble/SphericalFunctions.jl#41]
According to the documentation for `D_iterator`, the `D^ell` matrices that result from this should be indexed `D^{ell}[ell+mp+1, ell+m+1]`, however comparing with `D[WignerDindex(ell,mp,m)]` (where `D` is the full D vector), it appears that they are tranposed from one another, and the `D^ell` matrices actually correspond to `D^{ell}[ell+m+1, ell+mp+1]`

```
r = cos(π/6) + sin(π/6)*imy
D = D_matrices(r, 1)
d1 = [DD[WignerDindex(1,mp,m)] for mp in -1:1, m in -1:1] # indexed how the documentation states, d1[ell+mp+1, ell+m+1]
d2 = zeros(ComplexF64, (3,3))
for (l,d) in zip(1:1, D_iterator(DD,1,1))
    d2[:] = d[:]
end

d1 == d2 # returns false
d1 == transpose(d2) # returns true, showing that they are transposes of each other
```

**Comment by moble** [comment]
You're right.  This is very embarrassing.  Sorry for the slow reply; I thought this was just another difference in conventions, so I put this issue on the back burner.  But the code's behavior is clearly different from what the docstring says.  I failed to translate correctly from my python work.  🙄

I'm working on PR #47 to note the error in the docs and issue a warning whenever `D_iterator` or `d_iterator` is used.  I want to issue a patch release with this behavior so that anyone who may be using those functions will hopefully get the update and see the warning before their drone malfunctions or whatever.  An actual fix will have to wait until I have more time.  So for now, I'd say just avoid these functions — or use `Logging.NullLogger` to quiet the warnings when you do.

Thanks for reporting this.
